ui/signals.py

```diff
@@ -0,0 +1,16 @@
+"""
+ui model signals
+"""
+from django.db.models.signals import pre_delete
+from django.dispatch import receiver
+
+from ui.models import VideoFile, VideoThumbnail
+
+
+@receiver(pre_delete, sender=VideoFile)
+@receiver(pre_delete, sender=VideoThumbnail)
+def delete_s3_files(sender, **kwargs):  # pylint:disable=unused-argument
+    """
+    Make sure S3 files are deleted along with associated video file/thumbnail object
+    """
+    kwargs['instance'].delete_s3_objects()
```

Comment:
Any concern here about the length of this operation causing the request to time out and the object not end up completely cleaned up / deleted? If so, I'd maybe defer cleanup of the S3 assets to a celery task.